main.py

```
@@ -1134,0 +1,1230 @@
+import argparse
+import contextlib
+import hashlib
+import logging
+import re
+import time
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple
```

Ruff (0.8.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)